openupgrade_scripts/scripts/sale_loyalty/16.0.1.0/post-migration.py

```diff
@@ -80,15 +80,21 @@ def _generate_sale_order_coupon_points(env):
             coupon_id, order_id, points, create_uid, write_uid, create_date, write_date
         )
         SELECT
-            id AS coupon_id,
-            order_id,
-            points,
-            create_uid,
-            write_uid,
-            create_date,
-            write_date
-        FROM loyalty_card
-        WHERE order_id IS NOT NULL;
+            lc.id AS coupon_id,
+            lc.order_id,
+            lc.points,
+            lc.create_uid,
+            lc.write_uid,
+            lc.create_date,
+            lc.write_date
+        FROM loyalty_card lc
+        WHERE lc.order_id IS NOT NULL
+        AND NOT EXISTS (
```

Comment:
Instead of this, do a more performant `LEFT JOIN` with the table, and check if `socp.id IS NULL` as the WHERE condition.

Reply:
thanks!